src/Lighthouse.Tests.Benchmark/Program.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Configuration;
+using Tmds.Utils;
+using Universe.CpuUsage;
+
+namespace Lighthouse.Tests.Benchmark
+{
+    public static class Program
+    {
+        private const int TestLength = 30; // in seconds
+        private const int TestDelay = 5; // in seconds
+        private const int TestRepeat = 20;
+        private const int TestClusterSize = 10;
+        
+        private const string ActorSystemName = "CpuTest";
+        private const string LighthouseAddress = "127.0.0.1";
+        private const int LighthousePort = 54053;
+
+        private static readonly List<CpuUsage> Usages = new List<CpuUsage>();
+        private static readonly List<Process> Processes = new List<Process>();
+
+        public static async Task<int> Main(string[] args)
+        {
+            // ExecFunction hook
+            if (ExecFunction.IsExecFunctionCommand(args))
+                return ExecFunction.Program.Main(args);
+            
+            // Start lighthouse
+            var service = new LighthouseService(LighthouseAddress, LighthousePort, ActorSystemName);
+            service.Start();
+
+            var executor = new FunctionExecutor(o =>
+            {
+                o.StartInfo.RedirectStandardError = true;
+                o.OnExit = p =>
+                {
+                    if (p.ExitCode != 0)
+                    {
+                        var message =
+                            "Function execution failed with exit code: " +
+                            $"{p.ExitCode}{Environment.NewLine}{p.StandardError.ReadToEnd()}";
+                        throw new Exception(message);
+                    }
+                };
+            });
+
+            // Spin up cluster nodes
+            for (var port = LighthousePort + 1; port < LighthousePort + 1 + TestClusterSize; ++port)
+            {
+                Processes.Add(executor.Start(LaunchNode, new []
+                {
+                    LighthouseAddress,
+                    LighthousePort.ToString(),
+                    ActorSystemName,
+                    port.ToString()
+                }));
+            }
+
+            // Wait until things settles down 
+            await Task.Delay(TimeSpan.FromSeconds(TestDelay));

```

Comment:
Makes sense - wait until the cluster is formed.

Log2(N) + 1 + [some fudge factor] is how many seconds it should take a cluster to form, where N is your target cluster size, FYI.